structuralcodes/codes/ec2/shear.py

```diff
@@ -0,0 +1,444 @@
+    # 0.2fcd.
+    sigma_cp = Ned / Ac
+    if sigma_cp <= 0.0:
+        value = 1.0
+    elif sigma_cp <= 0.25 * fcd:
+        value = 1.0 + sigma_cp / fcd
+    elif sigma_cp <= 0.5 * fcd:
+        value = 1.25
+    elif sigma_cp < fcd:
+        value = 2.5 * (1 - sigma_cp / fcd)
+    else:
+        raise ValueError(
+            f"sigma_cp/fcd={sigma_cp/fcd}. Prestress has to be smaller"
+            " than design compressive strength."
+        )
+    return value
+
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+#                               NORMAL CONCRETE                               #
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+
+# Equation (6.2 a + b)
+def VRdc(
+    fck: float,
+    d: float,
+    Asl: float,
+    bw: float,
+    NEd: float,
+    Ac: float,
+    k1: float = 0.15,
+    gamma_c: float = 1.5,
+) -> float:
+    """Compute the design strength of the shear resistance.
+
+    EN 1992-1-1 (2005), Eq. (6.2)
+
+    Args:
+        fck (float): The characteristic compressive strength in MPa.
+        d (float): The effective depth of the cross-section in mm.
+        Asl (float): The cross-sectional area of the tensile
+            reinforcement, anchored atleast (lbd + d) beyond the
+            considdered cross-section, in mm2.
+        bw (float): The smallest width of the cross-section in tension
+            in mm.
+        NEd (float): The normal force in the cross-section due to
+            loading or prestress (NEd > 0 for compression) in N.
+        Ac (float): The cross-sectional area of the concrete in mm2.
+
+    Kwargs:
+        k1 (float): Factor used to include the effect of the normal
+            stress into the shear resistance of the concrete. Default
+            value = 0.15, value might differ between National Annexes.
+        gamma_c (float): Partial factor for concrete. Default value =
+            1.5, value might differ between National Annexes.
+
+    Returns:
+        float: The concrete shear resistance in MPa.
+    """
+    fcd = fck / gamma_c
+    CRdc = 0.18 / gamma_c
+    return (
+        max(
+            CRdc * _k(d) * (100 * _rho_L(Asl, bw, d) * fck) ** (1.0 / 3.0)
+            + k1 * _sigma_cp(NEd, Ac, fcd),  # VRdc
+            vmin(fck, d) + k1 * _sigma_cp(NEd, Ac, fcd),  # VRdcmin
+        )
+        * bw
+        * d
+    )
+
+
+# Equation (6.3N)
+def vmin(fck: float, d: float) -> float:
+    """Compute the minimum shear resistance of the concrete.
+
+    EN 1992-1-1 (2005), Eq. (6.3)
+
+    Args:
+        fck (float): The characteristic compressive strength in MPa.
+        d (float): The effective depth of the cross-section in mm.
+
+    Returns:
+        float: The minimal shear stress resistance of the concrete in
+            MPa.
+    """
+    return 0.035 * _k(d) ** (3.0 / 2.0) * fck ** (1 / 2)
+
+
+# Equation (6.4)
+def Vrdc_prin_stress():

```

Comment:
I have seen that this method (and others inside Eq. 6.4 from EC2) are not yet implemented `pass`.

Is there any cause why they cannot be implemented now? If not, it may be better to do it now in order to avoid revisiting our code later 🤔. 

In case it is not possible, do you consider it is a good idea to (at least) do one of the following?:

- Raise a runtime exception: `raise RuntimeError(msg)`
- Use a warning so we are not so strict: `warnings.warn('not implemented' ,RuntimeWarning)`

We have not discussed this issue yet but, if needed, we may ask to other contributor! 😁